apps/web/src/pages/templates/hooks/useTemplatesStoreModal.tsx

```diff
@@ -0,0 +1,21 @@
+import { useDisclosure } from '@mantine/hooks';
+
+import { useInlineComponent } from '../../../hooks';
+import { ITemplatesStoreModalProps, TemplatesStoreModal } from '../components/templates-store';
+
+const NULL_COMPONENT = () => null;
+
+export const useTemplatesStoreModal = () => {
+  const [isOpened, { open, close }] = useDisclosure(true);
+
+  const Component = useInlineComponent<ITemplatesStoreModalProps>(TemplatesStoreModal, {
+    isOpened,
+    onClose: close,
+  });
+
+  return {
+    TemplatesStoreModal: isOpened ? Component : NULL_COMPONENT,
+    openModal: open,
+    closeModal: close,
```

Comment:
What do you think about simplifying the name and removing the 'Modal'? because in the useTemplatesStoreModal context, it is self-explanatory IMO. 
And then the client of the hook could rename if needed

Reply:
the hook return modal, so I think it should specify that it's not a regular hook, but the one that returns the modal component